sky/serve/autoscalers.py

```diff
@@ -238,3 +264,168 @@ def _get_replica_ids_to_scale_down(num_limit: int) -> List[int]:
         if not scaling_options:
             logger.info('No scaling needed.')
         return scaling_options
+
+
+class SpotRequestRateAutoscaler(RequestRateAutoscaler):
+    """SpotRequestRateAutoscaler: Use spot to autoscale based on request rate.
+
+    This autoscaler uses spot instances to save cost.
+    """
+
+    NAME: Optional[str] = 'SpotRequestRateAutoscaler'
+
+    def __init__(self, spec: 'service_spec.SkyServiceSpec') -> None:
+        super().__init__(spec)
+        self.spot_placer: 'spot_policy.SpotPlacer' = (
+            spot_policy.SpotPlacer.from_spec(spec))
+
+    def handle_active_history(self, history: List[str]) -> None:
+        for zone in history:
+            self.spot_placer.handle_active(zone)
+
+    def handle_preemption_history(self, history: List[str]) -> None:
+        for zone in history:
+            self.spot_placer.handle_preemption(zone)
+
+    def _get_spot_resources_override_dict(self) -> Dict[str, Any]:
+        return {'use_spot': True, 'spot_recovery': None}
+
+    def _scale_spot_instances(
+        self, num_alive_spot: int,
+        launched_replica_infos: List['replica_managers.ReplicaInfo']
+    ) -> List[AutoscalerDecision]:
```

Comment:
Could we add docstr for those autoscaling functions to briefly describe its algorithm? The code is a little bit hard to read..